File: backend/planning-service/src/main/java/com/roomwise/planning/entity/Planning.java
package com.roomwise.planning.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalTime;

@Entity
@Data
@NoArgsConstructor
@Table (name="Planning")
@AllArgsConstructor
public class Planning {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDate date;
    private LocalTime heureDebut;
    private LocalTime heureFin;

    private String cours;
    private String proprietaireFiliere;
    private String proprietaireNiveau;

    private Long salleId; 
    private Long adminId;
}

File: frontend/frontend-admin/src/app/data/Planning.ts
import { SalleDeClasse } from "../api/apiServices";

export interface Planning {
    id: number; // Modifié pour correspondre au Long en Java, si vos IDs sont numériques
    date: string; // Correspond à LocalDate en Java (format "YYYY-MM-DD")
    heureDebut: string; // Correspond à LocalTime en Java (format "HH:MM" ou "HH:MM:SS")
    heureFin: string; // Correspond à LocalTime en Java (format "HH:MM" ou "HH:MM:SS")
    proprietaireFiliere: string; // Correspond à proprietaireFiliere
    proprietaireNiveau: string; // Correspond à proprietaireNiveau
    cours:string;
    salleId: number; // Correspond à Long salleId en Java
    adminId: number; // Correspond à Long adminId en Java
}

export type Status = "RESERVER" | "OCCUPEE" | "LIBRE";

export interface Salle {
    id: number; // Conserve string pour l'ID de Salle, comme dans vos données
    nom: string;
    Statut: Status;
}



export function getSalleIdByNom(salleName: string, salles: SalleDeClasse[]): number  {
    const salleTrouvee = salles.find(salle => salle.nom === salleName);
    return salleTrouvee ? salleTrouvee.id :0;
}